js/stringProcessing/extractUrl.js

```diff
@@ -0,0 +1,17 @@
+/**
+ * Retrieves all URLs from a string, based on the found anchors.
```

Comment:
Why would you try to match all href's in a URL, you are only throwing one anchor in this function so wouldn't it make more sense to only match once and just return '' or the matched url?